caproto/_state.py

```diff
@@ -171,32 +169,42 @@
 
             ClearChannelRequest: MUST_CLOSE,
             ServerDisconnResponse: CLOSED,
-            ErrorResponse: ERROR,
+            ErrorResponse: CONNECTED,
             # The commands ReadRequest, WriteRequest, WriteResponse, and
             # ReadSync (deprecated in 3.13) will need to be added here if we
             # want to support them.
         },
         MUST_CLOSE: {
             ClearChannelResponse: CLOSED,
             ServerDisconnResponse: CLOSED,
-            ErrorResponse: ERROR,
+            ErrorResponse: MUST_CLOSE,
         },
         CLOSED: {
             # a terminal state
         },
-        ERROR: {
+        FAILED: {
             # a terminal state
+            ClearChannelResponse: FAILED,
+            ServerDisconnResponse: FAILED,
+            ErrorResponse: FAILED
         },
     },
 }
 
 STATE_TRIGGERED_TRANSITIONS = {
     # (CHANNEL_STATE, CIRCUIT_STATE)
     CLIENT: {
-        (NEED_CIRCUIT, CONNECTED): (SEND_CREATE_CHAN_REQUEST, CONNECTED),
+        (SEND_CREATE_CHAN_REQUEST, DISCONNECTED): (CLOSED, DISCONNECTED),
```

Comment:
Potentially. I want to leave this here for now because it might be useful later and I'm not sure we're done fiddling with it.